### PR TITLE
Find unit by ID

### PR DESCRIPTION
**Pull request by DominicWatts**
Find unit by ID function

https://github.com/hitmeister/api-sdk-php/issues/79

### CODE REVIEW

File: src/Namespaces/UnitsNamespace.php [hunk 1]
```diff
@@ -68,7 +68,24 @@ public function findByEan($ean, $idOffer = null, $embedded = null, $limit = 30,
...
+    }
+    
+	/**
+	 * @param int|string $idOffer
+	 * @param string[]   $embedded
+	 * @param int        $limit
+	 * @param int        $offset
+	 * @return Cursor|UnitSellerTransfer[]
+	 */
+	public function find($idOffer = null, $embedded = null, $limit = 30, $offset = 0)
```
Comment (johnholiver) [inline src/Namespaces/UnitsNamespace.php]:
```suggestion
	public function findByIdOrder($idOffer, $embedded = null, $limit = 30, $offset = 0)
```

File: tests/Namespaces/UnitsNamespaceTest.php [hunk 2]
```diff
@@ -100,6 +100,48 @@ public function testFindByEan()
...
 		$result = iterator_to_array($result);
 
+		/** @var \Hitmeister\Component\Api\Transfers\UnitSellerTransfer[] $result */
+		$this->assertEquals(1, count($result));
+		$this->assertInstanceOf('\Hitmeister\Component\Api\Transfers\UnitSellerTransfer', $result[0]);
+		$this->assertEquals(350822415008, $result[0]->id_unit);
+		$this->assertEquals('new', $result[0]->condition);
+    }
+    
+    public function testFind()
```
Comment (johnholiver) [inline tests/Namespaces/UnitsNamespaceTest.php]:
```suggestion
    public function testFindByIdOffer()
```

File: tests/Namespaces/UnitsNamespaceTest.php [hunk 3]
```diff
@@ -100,6 +100,48 @@ public function testFindByEan()
...
+				'json' => [
+					[
+						'id_unit' => 350822415008,
+						'condition' => 'new'
+					]
+				]
+			]);
+
+		$namespace = new UnitsNamespace($this->transport);
+		$result = $namespace->find('136', ['item']);
```
Comment (johnholiver) [inline tests/Namespaces/UnitsNamespaceTest.php]:
```suggestion
		$result = $namespace->findByIdOffer('136', ['item']);
```

**Review by johnholiver — CHANGES_REQUESTED** [review]
Hello Dominic, thanks for submitting a PR.

In order for me to approve, I need to make sure that the changes follow a certain standard.
I've requested in this review for a refactor over the name of the method, from `find()` to `findByIdOrder`, making it more descriptive. Additionally, I've suggested removing the default null value for `$idOffer` as a complement to the specificity of the change above.

Finally, it could be understood that you wanted to make a generic find, for situations where no parameter is passed. But in that case, I'd recommend having yet another method without the filters ($idItem, $idOffer, or $ean) in the method signature.

**Comment by DominicWatts** [comment]
Updated to findByIdOffer

**Review by johnholiver — APPROVED** [review]

**Comment by DominicWatts** [comment]
Not sure if you are expecting me to do anything else

**Comment by NastiaNovik** [comment]
@DominicWatts  thank you for your contribution and sorry for the delay.

**Pull request merged**